k8s/production/runners/protected/graviton/2/release.yaml

```diff
@@ -45,6 +45,22 @@ spec:
     runners:
       config: |
         [[runners]]
+          pre_build_script = """
+          echo 'Executing Spack pre-build setup script'
+
+          python -m venv pre_build_venv
+          source pre_build_venv/bin/activate
+          python -m pip install -qr https://raw.githubusercontent.com/spack/spack-infrastructure/gitlab-pre-build-script/scripts/gitlab_runner_pre_build/requirements.txt
+          python -c "import requests;f=open('pre_build.py','w');f.write(requests.get('https://raw.githubusercontent.com/spack/spack-infrastructure/gitlab-pre-build-script/scripts/gitlab_runner_pre_build/pre_build.py').text);f.close()"
+          python pre_build.py > envvars
+
+          deactivate
+
+          source envvars
+          rm -rf pre_build_venv envvars
+          unset GITLAB_OIDC_TOKEN
```

Comment:
@scottwittenburg This is the snippet that we'll need to configure for external runners too, so it needs to be mostly unchanging to avoid coordination delays. Does this look reasonable to you in terms of being minimally invasive, cross platform/environment, and reliable (not bogging down pipelines or introducing flakiness)?

Reply:
I think it's wise to roll this out for just one type of runner for a little while before widely deploying it. I picked graviton2 arbitrarily.

Reply:
@eugeneswalker Do you see any problem adding this to your runner configs?  

It will allow us to stop putting the aws credentials into the gitlab ci variables, and even nicer, we should no longer need to worry about credentials expiring out from under jobs (meaning we could retry failed jobs, even if more than 12 hrs have passed since job creation).